app/controllers/CheckYourAnswersController.scala

```diff
@@ -64,7 +62,7 @@ class CheckYourAnswersController @Inject()(appConfig: FrontendAppConfig,
 
   def onSubmit(): Action[AnyContent] = (getData andThen requireData).async {
     implicit request =>
-      implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
+      implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
```

Comment:
The change from the deprecated use of `fromHeaderAndSession` can be updated in `DataRetrievalAction` also.